### PR TITLE
provider/kubernetes: Cache security groups

### PR DESCRIPTION
**Pull request by lwander**
Dont merge yet - security groups aren't showing up in deck (I'm guessing I have to surface the relationships between load balancers and security groups). 

@duftler

### CODE REVIEW

File: clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/cache/Keys.groovy [hunk 1]
```diff
@@ -53,6 +53,10 @@ class Keys {
       return null
     }
 
+    if (parts[0] != GoogleCloudProvider.GCE) {
```
Comment (lwander) [inline clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/cache/Keys.groovy]: Bug in GCE caching that was causing it to crash when other providers are enabled.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesSecurityGroup.groovy [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    this.application = application
+    this.accountName = account
+    this.region = ingress.metadata.namespace
+    this.namespace = this.region
+    this.name = ingress.metadata.name
+    this.id = this.name
+
+    this.inboundRules = (includeRules ? (ingress.spec.rules?.collect { rule ->
+      def defaultPort = new Rule.PortRange(startPort: 80, endPort: 80)
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/model/KubernetesSecurityGroup.groovy]: Couldn't hurt to extract these into constants.

**Comment by duftler** [comment]
LGTM.

**Comment by lwander** [comment]
@duftler updated to attach security groups to their server groups.

**Comment by duftler** [comment]
Still LGTM (except for the failing unit tests).

**Pull request merged**